src/plugins/data_source/server/client/configure_client.ts

```diff
@@ -69,24 +66,25 @@ export const configureClient = async (
       dataSource = await getDataSource(dataSourceId!, savedObjects);
     }
 
-    const rootClient = getRootClient(
-      dataSource,
-      openSearchClientPoolSetup.getClientFromPool,
-      dataSourceId
-    ) as Client;
+    let clientPool = openSearchClientPoolSetup;
+    const authenticationMethod = getAuthenticationMethod(dataSource, authRegistry);
+    if (authenticationMethod !== undefined) {
+      clientPool = authenticationMethod.clientPoolSetup;
+    }
```

Comment:
what is the type of clientPool? why do we need to override it from the `authenticationMethod`

Reply:
Type is [`OpenSearchClientPoolSetup`](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/plugins/data_source/server/client/client_pool.ts#L13).

Reply:
As clientpool from auth method removed, this code has been changes in latest revision.